tethys_apps/templatetags/site_settings.py

```diff
@@ -1,28 +1,55 @@
+import logging
+from pathlib import Path
+
 from django import template
 from django.template.defaultfilters import stringfilter
 from django.conf import settings
 
-from pathlib import Path
-
 from ..static_finders import TethysStaticFinder
 
 static_finder = TethysStaticFinder()
 
 register = template.Library()
 
+log = logging.getLogger(f"tethys.{__name__}")
+
 
 @register.filter
 @stringfilter
 def load_custom_css(var):
+    """Load Custom Styles defined in Tethys Portal -> Site Settings
+
+    Args:
+        var: a filename of CSS to load or CSS text to embed into the page
+
+    Returns:
+        a string of HTML that either embeds CSS text or points to a file
+
+    """
+    if not var.strip():
+        return ""
     if var.startswith("/"):
         var = var.lstrip("/")
 
-    if (Path(settings.STATIC_ROOT) / var).is_file() or static_finder.find(var):
-        return f'<link href="/static/{var}" rel="stylesheet" />'
+    try:
+        if (Path(settings.STATIC_ROOT) / var).is_file() or static_finder.find(var):
+            return f'<link href="/static/{var}" rel="stylesheet" />'
 
-    else:
         for path in settings.STATICFILES_DIRS:
             if (Path(path) / var).is_file():
                 return f'<link href="/static/{var}" rel="stylesheet" />'
+    except OSError as e:
```

Comment:
```suggestion
    # If the string is too long for a file path, which could happen if it is CSS, 
    # an OSError will be raised during the file path checks. This could also happen
    # if a lengthy file path is given or is otherwise invalid.
    except OSError as e:
```